packages/cli/src/bin/utils/getEndpointUrl.ts

```diff
@@ -0,0 +1,56 @@
+export type FusionEnv = 'ci' | 'fqa' | 'tr' | 'fprd';
+
+/**
+ * Retreive full endpoint URI to env in service-discovery
+ * @param endpoint <string> The endpoint to call in+ uri
+ * @param fusionEnv <FusionEnv> The Fusion env to get uri for
+ * @param service <string> Custom service uri to use insted of Fusion
+ * @param version <string> The version of the api to use
+ * @returns <string> The uri with endpoint
+ */
+export const getEndpointUrl = async (
+    endpoint: string,
+    fusionEnv: FusionEnv,
+    service: string,
+    version: string = '1.0',
+): Promise<string> => {
+    const { CUSTOM_APPAPI, FUSION_CLI_ENV, FUSION_TOKEN } = process.env;
+
+    /* use consumer provided api url */
+    if (service || CUSTOM_APPAPI) {
+        return service ?? CUSTOM_APPAPI;
+    }
+
+    /* Env has changed get new api url */
+    if (FUSION_CLI_ENV !== fusionEnv || !process.env.FUSION_CLI_APPAPI) {
+        process.env.FUSION_CLI_ENV = fusionEnv;
+
+        const requestService = await fetch(
+            `https://discovery.ci.fusion-dev.net/service-registry/environments/${fusionEnv}/services/apps`,
```

Comment:
use as doc

Reply:
Done

Reply:
adding a todo for now. we need to get it out as alpha and it works. can fix it before beta release.